costmap_2d/include/costmap_2d/layered_costmap.h

```diff
@@ -79,20 +79,24 @@ class LayeredCostmap
   void resizeMap(unsigned int size_x, unsigned int size_y, double resolution, double origin_x, double origin_y,
                  bool size_locked = false);
 
-  void getUpdatedBounds(double& minx, double& miny, double& maxx, double& maxy)
+  void getUpdatedBounds(double& minx, double& miny, double& maxx, double& maxy, double t = 0)
   {
-    minx = minx_;
-    miny = miny_;
-    maxx = maxx_;
-    maxy = maxy_;
+    int i;
+    if (timestep_ == 0)
+      i = 0;
+    else
+      i = std::min((int)timed_bounds_.size()-1, int(t/timestep_));
```

Comment:
Q: would make sense to round `t/timestep_`?

S: 
`const int i = timestep_ <= 0 ? 0 : std::clamp<int>(std::round(t/timestep_), 0, static_cast<int>(timed_bounds_.size()) - 1);`

Reply:
also consider making this a class method, since it's used below too, and in `getCostmap`